packages/utils/src/store/Store.ts

```diff
@@ -1,17 +1,46 @@
+import { useEventCallback } from '../useEventCallback';
+import { useStore } from './useStore';
+
 type Listener<T> = (state: T) => void;
 
+type MaybeCallable = (...args: any[]) => any;
+
+type ContextFunctionKeys<Context> = {
+  [Key in keyof Context]-?: Extract<Context[Key], MaybeCallable> extends never ? never : Key;
+}[keyof Context];
+
+type ContextFunction<Context, Key extends keyof Context> = Extract<Context[Key], MaybeCallable>;
+
 /**
  * A data store implementation that allows subscribing to state changes and updating the state.
  * It uses an observer pattern to notify subscribers when the state changes.
  */
-export class Store<State> {
+export class Store<
+  State,
+  Context = Record<string, never>,
+  Selectors extends Record<string, (state: State) => any> = Record<string, never>,
+> {
+  /**
+   * The current state of the store.
+   * This proprerty shouldn't be used directly. Use `useState` to subscribe to state changes.
+   */
   public state: State;
 
+  /**
+   * Non-reactive values such as refs, callbacks, etc.
+   * Unlike `state`, this property can be accessed directly.
+   */
+  public readonly context: Context;
+
   private listeners: Set<Listener<State>>;
 
-  constructor(state: State) {
+  private selectors?: Selectors;
```

Comment:
```suggestion
  private selectors: Selectors | undefined;
```

nitpick: The `?` notation implies the possibility of the field being absent, while `| undefined` implies the field is always there but possibly contains an `undefined` value. The latter is more correct.